### ISSUE TITLE
Support Twilio Webhooks? 

### ISSUE THREAD

**Issue by legrego** [legrego/HSPI_TwilioMessaging#16]
Currently, this plugin polls the Twilio API every X seconds to see if any messages have been received. Twilio supports webhooks that can notify consumers of messages without the need for polling.

## Benefits
- More immediate feedback of incoming messages. This would allow triggers to fire faster
- The plugin can respond to the webhook with a TwMl response, which would allow us to reply to the recipient.

## Drawbacks
- Requires HomeSeer to be publicly accessible -- Twilio needs to make an HTTP(s) call into HomeSeer to notify us of the incoming message. This potentially requires networking configuration on the end-users part, which isn't very friendly. UPnP is an option, but not something I want to explore for this plugin...it's too much of a security hazard.

**Comment by MattL0** [comment]
Not possible via myhs? I know phlocation does it with apple.

And Simplextechnologie have a plugin named ''MyHSToken''  in the beta section.  if Twillo need a real domain adress...maybe we can give them the my hs adres with a token created by MyHSToken.
?


Here is the forum for MyHSToken: https://forums.homeseer.com/forum/3rd-party-developer-area/general-developer-discussion/simplex-technology/1304696-beta-release-myhstoken

**Comment by legrego** [comment]
@MattL0 that's an interesting thought. I don't personally use myhs, so I hadn't considered that as an option. Would you be willing to test a beta version against myhs if I work on this?

**Comment by MattL0** [comment]
yes !

**Comment by mrizzo123** [comment]
I would also like this feature and would help test.

**Comment by legrego** [comment]
@mrizzo123 do you also use the "my homeseer" feature for remote access to your setup?

**Comment by mrizzo123** [comment]
> @mrizzo123 do you also use the "my homeseer" feature for remote access to your setup?

Yes, I use "my homeseer", however, I am sophisticated enough to handle an inbound NAT for a webhook if you want to start testing or implementing that way first.

**Comment by snakeeyes5** [comment]
Curious if this is still in the works? I would be willing to test also, and also wondering what your HS4 plans are at them moment?

**Comment by legrego** [comment]
This is something I'm still interested in tackling, I just haven't had the time to dedicate to this feature.

Same goes for HS4 support -- I bought an HS4 license, but I haven't provisioned an instance yet to see what would be involved to migrate this over.

**Comment by MattL0** [comment]
I dropped myhs altogether .  I use a hue emulator to control my devices ''locally'' . And an alexa package on nodered to control scenes. 

I would need to implement something to test with webhook. I just need to read a couple of how too lol. 

edit , also an hs4 user here.
